design-proposals/migration-target.md

```diff
@@ -0,0 +1,177 @@
+The motivation is to better define a boundary between what is an absolute and long-lasting property of a VM (like affinity) with what is just an optional property of the single migration attempt.
+This could also be relevant in terms of personas: we could have the VM owner/developer that is going to specify long-lasting affinity for a VM that is part of an application composed by different VMs and pods and a cluster admin/operator that needs to temporary override that for maintenance reasons.
+On the other side the VM owner is not required/supposed to be aware of node names. 
+
+## Goals
+- A user allowed to trigger a live-migration of a VM and list the nodes in the cluster is able to rely on a simple and direct API to try to live migrate a VM to a specific node.
+- The explict migration target overrules a nodeSelector or affinity and anti-affinity rules defined by the VM owner (still debated, see section [How to propagate the named node to the target virt-launcher pod](#how-to-propagate-the-named-node-to-the-target-virt-launcher-pod)).
+- The live migration then can successfully complete or fail for various reasons exactly as it can succeed of fail today for other reasons.
+- The target node that is explicitly required for the actual live migration attempt should not influence future live migrations or the placement in case the VM is restarted. For long-lasting placement, nodeSelectors or affinity/anti-affinity rules are the only way to go.
+
+## Non Goals
+- this proposal is not defining a custom scheduler plugin nor suggesting to alter how the default k8s scheduler works with `nodeName`, `nodeSelector` and `affinity/anti-affinity` rules. See https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/ for the relevant documentation
+
+## Definition of Users
+- VM owner: the user who owns a VM in his namespace on a Kubernetes cluster with KubeVirt
+- Cluster-admin: the administrator of the cluster
+
+## User Stories
+- As a cluster admin I want to be able to try to live-migrate a VM to specific node for various possible reasons such as:
+  - I just added to the cluster a new powerful node and I want to migrate a selected VM there without trying more than once according to scheduler decisions
+  - I'm not using any automatic workload rebalancing mechanism and I periodically want to manually rebalance my cluster according to my observations
+  - Foreseeing a peak in application load (e.g. new product announcement), I'd like to balance in advance my cluster according to my expectation and not to current observations
+  - During a planned maintenance window, I'm planning to drain more than one node in a sequence, so I want to be sure that the VM is going to land on a node that is not going to be drained in a near future (needing then a second migration) and being not interested in cordoning it also for other pods
```

Comment:
IIUC, a drain simply means applying a combination of `NoSchedule` and `NoExecute` taints on nodes. The first means that pods can still run on the node, but new pods won't be scheduled to it, while the second means that running pods will be evicted from the node.

If so, the admin can first apply the `NoSchedule` taint to all nodes that are planned for maintenance, and only then apply the `NoExecute` taint.

I think this addresses this story.